azavea/django-ecsmanage
# Improve tox, Travis, and local test / build configs

## Overview

This introduces a variety of local development fixes, Travis / tox optimizations and best practices, and drops support for end-of-life Python and Django versions.

Most of these updates reflect similar (and massively informative 💖) contributions made in https://github.com/azavea/django-amazon-ses/compare/3.0.0...96e53398aee537feab118c39312b40f22e5f390e.

* Drop support for Python 2.7 and 3.4
* Drop support for Django 2.0 and 2.1
* Fix Django `master` builds which would have previously resolved to a 3.x version
* Fix formatting / linting configs for `scripts/test`
* Fix source code formatting issues reported by `black`
* Move most settings from `settings.py` to `settings.cfg`
* Remove deprecated config options from `settings.py` / `settings.cfg`
* Remove redundant config options from `tox.ini` and `travis.yml`
* Run linting in separate `tox` environment so that it isn't run unnecessarily on every build

Closes #15 

### Notes

There's some formatting fixes here. You'll probably want to hide whitespace diffs.

## Testing Instructions

 * Review [Travis build logs](https://travis-ci.org/azavea/django-ecsmanage/builds/655979973). Verify that `TOXENV=lint` job runs.
 * Run `scripts/update` and `scripts/test` locally and confirm that all linting and tests pass.
